enhancements/installer/aws-load-balancer-type.md

```diff
@@ -0,0 +1,318 @@
+However when the default IngressController CR is deleted, default ingress controller is recreated but as an CLB/ELB.
+This  behavior is expected as of now as the ingress operator by default creates CLB for default ingress controller. 
+Also if the CR is deleted then the operator wont be able check the information whether NLB is needed or not. 
+So, we need to store this information in the cluster object of ingress.
+
+### Goals
+
+As an administrator, I would like to specify the load balancer type in AWS either to NLB or CLB/ELB during installation.
+As an user/administrator, when an ingress controller CR is deleted then I would like the ingress controller to be recreated with the load balancer mentioned during installation.
+In short I would like the persistence of the load balancer type in the default ingress controller.
+
+### Non-Goals
+
+N/A
+
+## Proposal
+
+We will be adding a field awsLBType in the install-config for the users to provide a load balancer either to NLB or CLB/ELB
+during installation.
+
+### Workflow Description
+
+The installer will accept a LB type from user and then we would set type of LB in the following object.
+
+`oc get ingress.config.openshift.io/cluster -o yaml`
+
+Then we would fetch info from this object and reconcile the operator to have the NLB changes reflected.
+
+### User Stories
+
+#### Story 1
+
+As an admin I want to install a specific load balancer rather than the default CLB/ELB.
+
+#### Story 2
+
+As an admin I want the load balancer type to persist even if the default ingress controller object is deleted which was mentioned during the installation.
+
+### API Extensions
+
+### Implementation Details/Notes/Constraints
+
+For the install config object we will be adding a field called awsLBType to the [API](https://github.com/openshift/installer/blob/master/pkg/types/aws/platform.go#L7)
+
+It will look as follows -
+```go
+package aws
+
+import "github.com/aws/aws-sdk-go/aws/endpoints"
+
+// Platform stores all the global configuration that all machinesets
+// use.
+type Platform struct {
+
+// LBType allows user to set a load balancer type.
+// Allowed values are nlb or elb
+// +optional
+LBType string `json:"lbType,omitempty"`
+}
+```
+
+In either the infra config cluster object or ingress config cluster object we will need to add api for AwsLBType which will store and persist the information
+of the desired LB type mentioned by the installer config API so that the ingress operator will refer the LB type from this object whenever someone accidentally deletes the ingress controller.
+Currently, the default LB type i.e CLB/ELB gets set if you delete the ingress controller object even when the desired the LB type for default ingress controller
+was set to NLB as ingress operator does not have a way to check the LB type as we don't store it.
+
+Option 1:(Recommended as it seems to be having all the cloud related information.)
```

Comment:
As an Installer reviewer, my vote is for Option 1 as it already seems to have Platform specific config in the the PlatformSpec and PlatformStatus types

Reply:
Based on https://github.com/openshift/installer/pull/6074 and https://github.com/openshift/api/pull/1209, it appears that the implementation is making progress with adding this field to the Infrastructure CR. Could you please update this proposal to reflect that and add Option 2 here to the "Althernatives" section.

Reply:
@sadasu done